Revamped Git-Tutorial: (#9)
frc1675/brownbook: controls/basics/how-to-git.md

## What is git and what is it used for? 

> <img src="../../assets/images/git-logo.png" alt="git logo" style="width:150px; float: right; padding: 25px;">

Author: Unfortunately, I have not yet found a good method of aligning the image in pure markdown. I found that through different markdown interpreters such as Kramdown, there exist ways of assigning ids and classes to elements. However, GitHub does not utilize Kramdown as their interpreter of choice. I also attempted to follow that blog post, and it lead me to dig a little deeper and found a [stack overflow thread]("https://stackoverflow.com/questions/255170/markdown-and-image-alignment") discussing how we could best resolve this. It seems the best option is still to use the bare HTML img tag as our interpreter doesn't seem to have the id assignment capability of other markdown interpreters. Another blog post seems to agree with using HTML tags for this purpose as in [this link]("https://talk.jekyllrb.com/t/proper-markdown-for-images-with-id-and-class/2035/3")  the OP comments that even Kramdown's method of assigning ids and classes looks messy and doesn't save much time. If you find a good method of doing this assignment, let me know and I'll implement it right away. For now, though, I believe that sticking with the bare HTML tag for the git-icon in particular is good practice!

> <img src="../../assets/images/git-logo.png" alt="git logo" style="width:150px; float: right; padding: 25px;">

Reviewer: OK. Lets leave it for now

> Git is something called a content tracker which is capable of cataloging all changes you make to tracked files. Git centers around the manipulation of branches of code, the merging and converging of existing code branches to improve a ending product. When you create or edit a file on your computer, the changes you make locally will be recorded and can differ from that of the global codebase otherwise known as the main branch. This dissonance provides greater quality control to the maintainer (permitting these changes), and allows for a more adaptable development cycle for the programmer through quality of life commands such as: "git revert", "git diff", and "git merge" How all of these benefits are achieved is through proper utilization.

Reviewer: ```suggestion
Git is a version control system which is capable of cataloging all changes you make to tracked files. Git centers around the manipulation of branches of code, to work separately on and eventually merge different sets of code to collaborate on a final product. When you create or edit a file on your computer, the changes you make locally can be recorded and can differ from that of the "global" code repository otherwise known as `origin` (for us, this one is hosted under the frc1675 organization). We use git with a specific set of commands and GitHub interactions to (mostly) enforce a process that provides a higher level of quality assurance via review and testing before new code joins the rest. How all of these benefits are achieved is through proper utilization.
```

### How to use git

Commonly Used Commands:
<ul>
    <li>git add %filename%</li>

>     The git add command adds a filename to the tracker and its changes will be recorded by git. Only files added through the git add command will be included in commits and pushes later down the line.

Reviewer: surround all things that would be put into a command line with backticks `like this`, provides easy dilneation as to what parts are the commands.

```suggestion
    The `git add` command adds a filename to the tracker and its changes will be recorded by git. Only files added through `git add` will be included in commits (and by implication, pushes) later down the line.
```

### How to use git

Commonly Used Commands:

* `git add %filename%`

    The `git add` command adds a filename to the tracker and its changes will be recorded by git. Only files added through the `git add` command will be included in commits and pushes later down the line.
* `git commit -m %user message%`

    The git commit command will record all changes that you make locally on your device. Said commit will be provided a unique code which can be referenced in other commands ie: revert, push, diff. All commits must contain a message, when writing the description try to be as descriptive as possible while also not overwhelming the reviewer.
* `git push origin %branchname%`

    How we stage changes in most content tracked codebases is through the introduction of new branches containing our code. The git push command takes our locally committed changes and "pushes" them to the external server connected to the repository.

![git development cycle](../../assets/images/git-cycle.png)

> For more info on commands and git, type: [`man git`]("https://git-scm.com/docs") into your command line. Or watch this [video]("https://www.youtube.com/watch?v=HkdAHXoRtos").

Reviewer: the quotes on these links are breaking them

### Create a GitHub account

> GitHub is one such online git service provider. To create a new GitHub account follow the instructions below or watch [video](https://www.youtube.com/watch?v=HkdAHXoRtos)!

Reviewer: ```suggestion
GitHub is one such online git service provider. To create a new GitHub account follow the instructions below or watch this [video](https://www.youtube.com/watch?v=HkdAHXoRtos)!
```
